pandas/tests/groupby/test_groupby.py

```diff
@@ -433,6 +433,33 @@ def test_frame_groupby_columns(tsframe):
         assert len(v.columns) == 2
 
 
+@pytest.mark.parametrize(
+    "func, args",
+    [
+        ("sum", []),
+        ("prod", []),
+        ("min", []),
+        ("max", []),
+        ("nth", [0]),
+        ("last", []),
+        ("first", []),
+    ],
+)
+def test_frame_groupby_avoids_mutate(func, args):
+    # GH28523
+    df = pd.DataFrame({"A": ["foo", "bar", "foo", "bar"], "B": [1, 2, 3, 4]})
+    grouped = df.groupby("A")
+
+    expected = grouped.apply(lambda x: x)
+
+    fn = getattr(grouped, func)
+    fn(*args)
```

Comment:
If `nth` is the only one that requires an argument I think OK to special case within the test

Reply:
@WillAyd the only other instance I looked at was inside of `_make_wrapper`, but it didn't seem related to this bug, and I haven't looked at it in depth.